src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDynamicDocumentInfoProviderBase.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    /// <summary>
+    /// Base MEF component for implementing a dynamic document info provider
+    /// Currently only used by VS Mac
+    /// </summary>
+    internal abstract class RazorDynamicDocumentInfoProviderBase
```

Comment:
I'd probably stick this in the Editor.Razor assembly as well. I can add onto the PR to make the VS impl use this.